languages/bash.scm

```diff
@@ -1,2 +1,114 @@
 ; Configuration
 (#language! bash)
+
+; Don't modify string literals
+(string) @leaf
+
+; Allow blank line before
+; TODO Add directives here...
+[
+  (comment)
+  (if_statement)
+] @allow_blank_line_before
+
+; Prepend spaces
+; (Not "surround", to avoid extraneous spaces between delimiters)
+; TODO Add keywords, etc., here...
+[
+  "if"
+  "then"
+  "elif"
+  "else"
+  "fi"
+  (string)
+  (test_command)
+] @prepend_space
+
+;; Commands
+
+; One command per line, modulo many exceptions:
+; * Strings of && and ||
+; * Pipelines
+; * Subshells
+; * As arguments to if, while, for, etc.
+; * Others?...
+;
+; FIXME Can this be generalised, or does *every* context need to be
```

Comment:
Maybe simply `(command) @prepend_spaced_softline`?

Reply:
This breaks, e.g., `if`-clauses. It will turn:

```bash
if something
```

into:

```bash
if
something
```

The same will happen in the (many) other contexts where commands are embedded into some clause.

Reply:
Got it.